### DISCUSSION TITLE
Cannot find matching entity for condition" when using upsertMany

### DISCUSSION THREAD

**Discussion by KaneGZ** [mikro-orm/mikro-orm#5512] [General]
### Describe the bug

A "Cannot find matching entity for condition" error gets raised when using upsertMany
![image](https://github.com/mikro-orm/mikro-orm/assets/51891452/b47778e7-4dc5-4225-af2f-9dd8385e4a90)

### Reproduction

```
import type { EntityProperty, Platform } from '@mikro-orm/mysql';

import { Type } from '@mikro-orm/mysql';
import { BigNumber } from 'bignumber.js';

export class BigNumberType extends Type<BigNumber, string> {
  static DEFAULT_PRECISION = 34;
  static DEFAULT_DECIMAL_PLACE = 18;

  convertToDatabaseValue(value: BigNumber): string {
    const { scale = BigNumberType.DEFAULT_DECIMAL_PLACE } = this.prop ?? {};
    return BigNumber(value).dp(scale, BigNumber.ROUND_DOWN).toFixed();
  }

  convertToJSValue(value: string): BigNumber {
    return new BigNumber(value);
  }

  toJSON(value: BigNumber, _platform: Platform): string {
    this.platform?.getConfig();
    return value.toFixed();
  }

  getColumnType(prop: EntityProperty, _platform: Platform): string {
    return `numeric(${prop.precision ?? BigNumberType.DEFAULT_PRECISION},${prop.scale ?? BigNumberType.DEFAULT_DECIMAL_PLACE})`;
  }
}

// Definitions
@Entity({
  tableName: 'accounts',
})
export class Account extends BaseEntity<'balance' | 'locked'> {
  [PrimaryKeyProp]?: ['member', 'currency'];

  @ManyToOne({
    primary: true,
    entity: () => Member,
    ref: true,
  })
  member!: Ref<Member>;

  @ManyToOne({
    primary: true,
    entity: () => Currency,
    ref: true,
  })
  currency!: Ref<Currency>;

  @Property({
    type: BigNumberType,
    default: 0,
  })
  balance: BigNumber = new BigNumber(0);

  @Property({
    type: BigNumberType,
    default: 0,
  })
  locked: BigNumber = new BigNumber(0);
}

// For context
await db.em.upsertMany(
  Account,
  currencies.map((currency) => {
    return {
      member,
      currency,
    };
    // OR
    // return {
    //   member: member.id,
    //   currency: currency.id,
    // };
  }),
  {
    onConflictAction: 'ignore',
  },
);
```

### What driver are you using?

@mikro-orm/mysql

### MikroORM version

6.2.1

### Node.js version

v20.12.1

### Operating system

Linux

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by KaneGZ** [comment]
There is a problem with the way bigint is handled, the member here is a number when it should be a string

File: node_modules/@mikro-orm/core/EntityManager.js
![image](https://github.com/mikro-orm/mikro-orm/assets/51891452/091389d4-9379-459e-bb3b-7f0b21fd4f45)

![image](https://github.com/mikro-orm/mikro-orm/assets/51891452/9cbaeb19-0197-4b45-8f76-ab8c5aa94fb8)

**Comment by B4nan** [comment]
What is BigNumberType? Please provide a complete reproduction.

**Reply by KaneGZ** [reply]
Here, It seems there is no problem with the custom type, the error is in the member(member_id:bigint) property https://github.com/mikro-orm/mikro-orm/discussions/5512#discussioncomment-9237429:
```
import type { EntityProperty, Platform } from '@mikro-orm/mysql';

import { Type } from '@mikro-orm/mysql';
import { BigNumber } from 'bignumber.js';

export class BigNumberType extends Type<BigNumber, string> {
  static DEFAULT_PRECISION = 34;
  static DEFAULT_DECIMAL_PLACE = 18;

  convertToDatabaseValue(value: BigNumber): string {
    const { scale = BigNumberType.DEFAULT_DECIMAL_PLACE } = this.prop ?? {};
    return BigNumber(value).dp(scale, BigNumber.ROUND_DOWN).toFixed();
  }

  convertToJSValue(value: string): BigNumber {
    return new BigNumber(value);
  }

  toJSON(value: BigNumber, _platform: Platform): string {
    this.platform?.getConfig();
    return value.toFixed();
  }

  getColumnType(prop: EntityProperty, _platform: Platform): string {
    return `numeric(${prop.precision ?? BigNumberType.DEFAULT_PRECISION},${prop.scale ?? BigNumberType.DEFAULT_DECIMAL_PLACE})`;
  }
}
```
